libs/report/ingestion/src/lib/ingestion.ts

```diff
@@ -0,0 +1,39 @@
+import { type ArgumentsCamelCase, type CommandModule } from 'yargs';
+import { askConsent, promptClientName, promptToProceedUploadFile } from './prompts';
+import { findManifestFile, getClientToken, sendManifest } from './send-manifest';
+import { type Options } from './types';
+
+export const reportIngestionCommand: CommandModule<object, Options> = {
+  command: 'generate',
+  describe: 'send manifest files information',
+  aliases: ['gen', 'g'],
+  builder: {
+    consent: {
+      describe: 'I understand that sensitive data may be uploaded to the server',
+      required: false,
+      default: false,
+      boolean: true,
+    },
+  },
+  handler: run,
+};
+
+async function run(args: ArgumentsCamelCase<Options>): Promise<void> {
+  const consent = await askConsent(args);
```

Comment:
FWIW, I find the current process wrt consent somewhat confusing and inconsistent.

Why do we need to have two consent prompts and also why only suppress the one of them if the `--consent` option is passed?

I suggest the following:
1. Look for the manifest file (without concerning ourselves with consent first).
2. If they have passed the `--consent` option, upload the data.
3. If the have not passed the `--consent` option, prompt them for consent to upload the specific file we have found (so that they have a chance to review it if needed).
4. If they give consent, upload the data.

```mermaid
flowchart TB
  A[Start] --> B
  B[Find manifest file] --> C
  C{'--consent' option?} -- Yes --> D
  D[Upload data] --> E[End]
  C -- No --> F
  F[Prompt for consent] --> G
  G{Given consent?} -- Yes --> D
  G -- No --> E
```

Reply:
It's not about asking consent twice, is that in case of multiple manifest found the user can choose which one to send , and avoid sending the wrong one by mistake. I could immagine in a php project there might be pckage.json or composer.json. So asking the user to confirm he wants to send that specific package makes sense to me.